### PR TITLE
MAINT: Simplify np.int_ inheritance

### PR DESCRIPTION
**Pull request by sethtroisi**
Broke out another PY3K change.

### CODE REVIEW

File: numpy/core/src/multiarray/multiarraymodule.c [hunk 1]
```diff
@@ -4235,35 +4235,13 @@ setup_scalartypes(PyObject *NPY_UNUSED(dict))
...
-#define INHERIT_INT(child, parent2)                                     \
-    SINGLE_INHERIT(child, parent2);
-#else
-#define INHERIT_INT(child, parent2)                                     \
-    Py##child##ArrType_Type.tp_flags |= Py_TPFLAGS_INT_SUBCLASS;        \
-    DUAL_INHERIT(child, Int, parent2);
-#endif
-
-#if defined(NPY_PY3K)
-#define DUAL_INHERIT_COMPARE(child, parent1, parent2)
```
Comment (sethtroisi) [inline numpy/core/src/multiarray/multiarraymodule.c]: This is no longer needed in PY3K

File: numpy/core/src/multiarray/multiarraymodule.c [hunk 2]
```diff
@@ -4235,35 +4235,13 @@ setup_scalartypes(PyObject *NPY_UNUSED(dict))
         return -1;                                                      \
     }
 
-/*
- * In Py3K, int is no longer a fixed-width integer type, so don't
- * inherit numpy.int_ from it.
- */
-#if defined(NPY_PY3K)
-#define INHERIT_INT(child, parent2)                                     \
-    SINGLE_INHERIT(child, parent2);
```
Comment (sethtroisi) [inline numpy/core/src/multiarray/multiarraymodule.c]: s/INHERIT_INT/SINGLE_INHERIT/ below leads to 

if SIZE = OTHER_SIZE:
  SINGLE_INHERIT(a, b)
else:
  SINGLE_INHERIT(a, b)

which can be simplified

File: numpy/core/src/multiarray/multiarraymodule.c [hunk 3]
```diff
@@ -4277,19 +4255,11 @@ setup_scalartypes(PyObject *NPY_UNUSED(dict))
...
     SINGLE_INHERIT(Short, SignedInteger);
 
-#if NPY_SIZEOF_INT == NPY_SIZEOF_LONG
-    INHERIT_INT(Int, SignedInteger);
-#else
     SINGLE_INHERIT(Int, SignedInteger);
-#endif
 
-    INHERIT_INT(Long, SignedInteger);
+    SINGLE_INHERIT(Long, SignedInteger);
```
Comment (seberg) [inline numpy/core/src/multiarray/multiarraymodule.c]:
```suggestion
```

File: numpy/core/src/multiarray/multiarraymodule.c [hunk 4]
```diff
@@ -4277,19 +4255,11 @@ setup_scalartypes(PyObject *NPY_UNUSED(dict))
     SINGLE_INHERIT(Byte, SignedInteger);
     SINGLE_INHERIT(Short, SignedInteger);
 
-#if NPY_SIZEOF_INT == NPY_SIZEOF_LONG
-    INHERIT_INT(Int, SignedInteger);
-#else
     SINGLE_INHERIT(Int, SignedInteger);
-#endif
```
Comment (seberg) [inline numpy/core/src/multiarray/multiarraymodule.c]:
```suggestion
```

File: numpy/core/tests/test_multiarray.py [hunk 5]
```diff
@@ -346,17 +346,8 @@ def test_int_subclassing(self):
...
-            assert_equal(isinstance(numpy_int, int), False)
-        else:
-            # Otherwise, it should inherit from int...
-            assert_equal(isinstance(numpy_int, int), True)
-
-            # ... and fast-path checks on C-API level should also work
-            from numpy.core._multiarray_tests import test_int_subclass
-            assert_equal(test_int_subclass(numpy_int), True)
+        # int_ doesn't inherit from int, because it's not fixed-width
+        assert_equal(isinstance(numpy_int, int), False)
```
Comment (seberg) [inline numpy/core/tests/test_multiarray.py]:
```suggestion
        assert not isinstance(numpy_int, int)
```
But its good as is, should not do too much general cleanup.

**Review by seberg — APPROVED** [review]
LGTM, thanks. Can apply the suggestions or not if you prefer (although I guess removing the empty lines is cleaner now).

File: numpy/core/src/multiarray/multiarraymodule.c (hunk 4 above)
Comment (eric-wieser) [reply numpy/core/src/multiarray/multiarraymodule.c]: ???

Comment (eric-wieser) [reply numpy/core/src/multiarray/multiarraymodule.c]: Ah, remove blank lines

Comment (seberg) [reply numpy/core/src/multiarray/multiarraymodule.c]: opst, that was supposed to remove the blank line

Comment (seberg) [reply numpy/core/src/multiarray/multiarraymodule.c]: wait it should, now that its just one line, I think its good to group the ints OTOH, should also remove the one above then.

Comment (eric-wieser) [reply numpy/core/src/multiarray/multiarraymodule.c]: I think github is just rendering this confusingly, I think the commit suggestion button will do the right thing

File: numpy/core/src/multiarray/multiarraymodule.c [hunk 6]
```diff
@@ -4277,19 +4255,11 @@ setup_scalartypes(PyObject *NPY_UNUSED(dict))
     SINGLE_INHERIT(Byte, SignedInteger);
     SINGLE_INHERIT(Short, SignedInteger);
```
Comment (seberg) [inline numpy/core/src/multiarray/multiarraymodule.c]:
```suggestion
```

File: numpy/core/tests/test_multiarray.py (hunk 5 above)
Comment (sethtroisi) [reply numpy/core/tests/test_multiarray.py]: I changed it to the form all other tests use.

Comment (sethtroisi) [reply numpy/core/tests/test_multiarray.py]: Thanks for all the suggestions.

**Comment by seberg** [comment]
Thanks!

**Pull request merged**